operator/webhook/introute/sync.py

```diff
@@ -388,9 +390,51 @@ def _new_deployment(parent):
 
     return deployment
 
+def _new_service(parent):
+    parent_metadata = parent["metadata"]
+
+    has_tls = _has_tls(parent)
+    scheme = _get_scheme(has_tls)
+    management_port = _get_management_port(has_tls)
+
+    service = {
+        "apiVersion": "v1",
+        "kind": "Service",
+        "metadata": {
+            "labels": {
+                "integration-route": parent_metadata["name"],
+                "prometheus-metrics-enabled": "true"
+            },
+            "name": parent_metadata["name"]
+        },
+        "spec": {
+            "ports": [
+                {
+                    "name": scheme,
+                    "port": management_port,
+                    "protocol": "TCP",
+                    "targetPort": management_port
+                }
+            ],
+            "selector": {
+                "app.kubernetes.io/instance": f'integrationroute-{parent_metadata["name"]}'
+            }
+        }
+    }
+
+    return service
+
+def _has_tls(parent) -> bool:
+    return "tls" in parent["spec"] and "keystore" in parent["spec"]["tls"]
+
+def _get_scheme(has_tls) -> str:
+    return "https" if has_tls else "http"
+
+def _get_management_port(has_tls) -> int:
+    return HTTPS_PORT if has_tls else HTTP_PORT
 
 def _gen_children(parent) -> List[Mapping]:
-    return [_new_deployment(parent)]
+    return [_new_deployment(parent), _new_service(parent)]
```

Comment:
Since the webhook is now creating a service as well, we should probably add that to the `childResources` list in the `integrationroute` `CompositeController` manifest.

Reply:
Fixed